todo_app/data/mongodb_requests.py

```diff
@@ -0,0 +1,45 @@
+import os
+from datetime import datetime
+from typing import Dict, Union
+
+import pymongo
+
+from todo_app.data.items import Item, ItemView, Status
+
+
+class MongoDbRequests:
+    def __init__(self):
+        client = pymongo.MongoClient(os.getenv("MONGODB_CONNECTION_STRING"))
+        db = client[os.getenv("DB_NAME")]
+        self.collection = db["items"]
+
+    def _json_to_item(self, json: Dict[str, Union[str, datetime]]) -> Item:
+        return Item(
+            id_=json["_id"],
+            status=Status(json["status"]),
+            title=json["title"],
+            description=json["description"],
+            last_modification=json["last_modified"],
+            due=datetime.fromisoformat(json["due"]) if json["due"] else None,
+        )
+
+    def get_items(self) -> ItemView:
+        return ItemView([self._json_to_item(item) for item in self.collection.find()])
+
+    def add_item(self, title: str, description: str, due: str) -> None:
+        self.collection.insert_one(
+            {
+                "title": title,
+                "status": Status.NOT_STARTED.value,
+                "description": description,
+                "due": due,
+                "last_modified": datetime.now(),
+            }
+        )
+
+    def remove_item(self, id_: str) -> None:
+        print(id_)
+        self.collection.delete_one({"_id": id_})
```

Comment:
I don't think this is working - certainly not for my testing. That seems to be because the ids aren't matching; [Mongo uses a type called "ObjectId"](https://pymongo.readthedocs.io/en/stable/api/bson/objectid.html) which we're comparing directly a string here; instead we'll want something like:
```suggestion
    def remove_item(self, id_: str) -> None:
        print(id_)
        self.collection.delete_one({"_id": ObjectId(id_)})
```

With ObjectId imported from `bson`.

Something similar for your updates is also required I believe

Reply:
Oh thats weird, it looks like this works using the cosmos db, but not the docker hosted database